dart/test/monster_test_my_game.example_generated.dart

```diff
@@ -643,6 +865,54 @@ class Stat {
   String toString() {
     return 'Stat{id: $id, val: $val, count: $count}';
   }
+
+  StatT unPack() => StatT(
+      id: id,
+      val: val,
+      count: count);
+
+  static int pack(fb.Builder fbBuilder, StatT object) {
+    if (object == null) return 0;
+    return object.pack(fbBuilder);
+  }
+}
+
+class StatT {
+  String id;
+  int val;
+  int count;
+
+  StatT({
+      this.id,
+      this.val,
+      this.count});
+
+  int pack(fb.Builder fbBuilder) => _pack(fbBuilder, HashMap<Object, int>());
+
+  /// Write the object to the [Builder] only once and reuse the same offset on
+  /// subsequent references of the same object.
+  ///
+  /// Note that this method assumes you call it using the same [Builder]
+  /// instance every time. The returned offset is only good for the [Builder]
+  /// used in the first call to this method.
+  int packOnce(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    return existingOffsets[this] ??= _pack(fbBuilder, existingOffsets);
+  }
+
+  int _pack(fb.Builder fbBuilder, Map<Object, int> existingOffsets) {
+    assert(existingOffsets != null);
+    assert(fbBuilder != null);
+    final int idOffset = fbBuilder.writeString(id);
+
+    fbBuilder.startTable();
+    if (idOffset != null) {
```

Comment:
compares integer with `null` ?

Reply:
ints are nullable in dart, unless you're in null-safety mode in which case it'd be written as `int?`.  

I don't think we should ever expect writeString to return null - for example if the buffer is out of space wouldn't we just throw?

Reply:
`writeString()` returns null if the given string (value) is null itself.